truffle.js

```diff
@@ -26,8 +10,22 @@ module.exports = {
     mainnet: {
       ref: 'mainnet-prod',
       network_id: 1,
-      provider: () => keystoreProvider(mainnetUrl),
+      provider: () => {
+        return new HDWalletProvider(
+          [process.env.PRIVATE_KEY],
+          'https://eth-mainnet.gateway.pokt.network/v1/<APP_ID>');
+      },
       gasPrice: 35000000000
```

Comment:
Why are we setting the gas price manually for mainnet? Shouldn't we be able to figure it out at runtime? If not, I'd put both this and the rinkeby gas price as a constant above, like `MAINNET_GAS_PRICE` and `RINKEBY_GAS_PRICE`

Reply:
> Why are we setting the gas price manually for mainnet? Shouldn't we be able to figure it out at runtime? If not, I'd put both this and the rinkeby gas price as a constant above, like `MAINNET_GAS_PRICE` and `RINKEBY_GAS_PRICE`

It's not common to create scripts that use APIs for fetching gas prices. It's not even ideal if who deploys the contract knows what they are doing.

Reply:
Yeah of course, I think it makes sense with volatile gas prices right now, but Ethers does actually a pretty good job of figuring out gas prices... Even though this is using web3, so, makes sense 😆